### DISCUSSION TITLE
What should I do to make Meta tags work Reactively with social media?

### DISCUSSION THREAD

**Discussion by arnonrdp** [quasarframework/quasar#12306] [General - Components / Directives / etc] [answered]
I was reading the following snippet from the documentation: https://quasar.dev/quasar-plugins/meta#reactive

And I was wondering how I should declare Open Graph and Twitter meta tags since reactive tags are not declared as objects.

Any idea?

**Comment by rstoenescu** [comment, marked as answer]
Make the function return your Meta state, whatever that is. It essentially acts like a computed prop and can contain the regular Meta state, but you can use other computed props as values.

**Reply by arnonrdp** [reply]
So, do you consider this is correct?
```js
<script>
import { useMeta } from "quasar";
import { useI18n } from "vue-i18n";
import { ref } from "vue";

export default {
  components: { Header },
  setup() {
    const { t } = useI18n();
    const description = ref("");
    const keywords = ref("");
    const ogDescription = ref("");
    const twitterDescription = ref("");
    useMeta(() => ({
      meta: {
        description: description.value,
        keywords: keywords.value,
        ogDescription: ogDescription.value,
        twitterDescription: twitterDescription.value,
      },
    }));
    function setLanguage() {
      description.value = { name: "description", content: t("meta.description") };
      keywords.value = { name: "keywords", content: t("meta.keywords") };
      ogDescription.value = { name: "og:description", content: t("meta.description") };
      twitterDescription.value = { name: "twitter:description", content: t("meta.description") };
    }
    return {
      setLanguage,
    };
  },
};
</script>
```

**Reply by rstoenescu** [reply]
Yes. But just to be clear: you don't require to manually trigger a function call. You could just use the `t()` calls inside of the useMeta() function.

**Reply by arnonrdp** [reply]
I am doing this, but popular meta tags websites doesn't recognize it.

If you check the website the meta tags are there: https://qa-livrero.vercel.app/

But when you try to insert into [MetaTags.io](https://metatags.io/) or [LinkedIn Post Inspector](https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fqa-livrero.vercel.app%2F) it doesn't work.
